lms/services/canvas_api/client.py

```diff
@@ -40,14 +43,22 @@ class CanvasAPIClient:
         Canvas API request fails for any other reason
     """
 
-    def __init__(self, authenticated_client, file_service):
+    def __init__(
+        self,
+        authenticated_client,
+        file_service: FileService,
+        with_folders: bool = False,
```

Comment:
This could be a parameter on the list_files method it self but we already have a reference to the AI in the factory, it would be more awkward the other way around.